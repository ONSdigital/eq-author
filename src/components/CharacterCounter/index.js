import React from "react";
import PropTypes from "prop-types";

const getLength = value => (value ? value.length : 0);

const CharacterCounter = ({ value, limit }) => (
  <span>Characters remaining: {limit - getLength(value)}</span>
);

CharacterCounter.propTypes = {
  value: PropTypes.string,
  limit: PropTypes.number.isRequired
};

export default CharacterCounter;
