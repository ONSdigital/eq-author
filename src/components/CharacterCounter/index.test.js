import React from "react";
import { shallow } from "enzyme";
import CharacterCounter from "components/CharacterCounter/index";

const createWrapper = (props = {}, render = shallow) => {
  return render(<CharacterCounter {...props} />);
};

describe("CharacterCounter", () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      value: "FooBar",
      limit: 25
    };

    wrapper = createWrapper(props);
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should update counter based on length of value", () => {
    wrapper = createWrapper({
      ...props,
      value: "FooBarFooBarFooBar"
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("should update counter with negative count when limit exceeded", () => {
    wrapper = createWrapper({
      ...props,
      value: "FooBarFooBarFooBarFooBarFooBarFooBar"
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("should handle null value", () => {
    wrapper = createWrapper({
      ...props,
      value: null
    });
    expect(wrapper).toMatchSnapshot();
  });
});
