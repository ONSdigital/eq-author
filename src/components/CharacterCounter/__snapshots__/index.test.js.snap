// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCounter should handle null value 1`] = `
<span>
  Characters remaining: 
  25
</span>
`;

exports[`CharacterCounter should render 1`] = `
<span>
  Characters remaining: 
  19
</span>
`;

exports[`CharacterCounter should update counter based on length of value 1`] = `
<span>
  Characters remaining: 
  7
</span>
`;

exports[`CharacterCounter should update counter with negative count when limit exceeded 1`] = `
<span>
  Characters remaining: 
  -11
</span>
`;
